cmake/modules/boards.cmake

```diff
@@ -61,43 +61,44 @@ if(NOT unittest IN_LIST Zephyr_FIND_COMPONENTS)
   list(APPEND BOARD_ROOT ${ZEPHYR_BASE})
 endif()
 
-string(FIND "${BOARD}" "@" REVISION_SEPARATOR_INDEX)
-string(FIND "${BOARD}" "/" IDENTIFIER_SEPARATOR_INDEX)
-
-if(NOT (REVISION_SEPARATOR_INDEX EQUAL -1 OR IDENTIFIER_SEPARATOR_INDEX EQUAL -1))
-  if(REVISION_SEPARATOR_INDEX GREATER IDENTIFIER_SEPARATOR_INDEX)
-    message(FATAL_ERROR "Invalid revision / identifier format, format is: "
-                        "<board>@<revision>/<identifier>"
+# Helper function for parsing a board's name, revision, and identifier,
+# from one input variable to three separate output variables.
+function(parse_board_components board_in name_out revision_out identifier_out)
+  if(NOT ${board_in} MATCHES "^([^@/]+)(@[^@/]+)?(/[^@]+)?$")
+    message(FATAL_ERROR "Invalid revision / identifier format for ${board_in}. "
```

Comment:
Printing the variable name here is sure a good thing. Especially for the var `BOARD` this will give the user a hint on what was wrong. 👍 

But to help extra for errors in alias.cmake / deprecated.cmake, should we consider to print the wrong value in addition ?
After all, the user will not know the value of `plank_DEPRECATED` without finding the right file.
For example: `Invalid revision / identifier format for plank_DEPRECATED (plonk/foo@1.0.0).` will be a bit more helpful for users who specified `-DBOARD=plank`

Non blocking comment.